0.59 changelog (#18575)
microsoft/PowerToys: README.md

- Windows 11 or Windows 10 v1903 (18362) or newer.
- Our installer will install the following items:

>    - [.NET 6.0.5 Desktop Runtime](https://dotnet.microsoft.com/en-us/download/dotnet/6.0#runtime-desktop-6.0.5) or a newer 6.0.x runtime. 

Reviewer: ```suggestion
   - [.NET 6.0.5 Desktop Runtime](https://dotnet.microsoft.com/download/dotnet/6.0#runtime-desktop-6.0.5) or a newer 6.0.x runtime. 
```

   - [Microsoft Edge WebView2 Runtime](https://go.microsoft.com/fwlink/p/?LinkId=2124703) bootstrapper. This will install the latest version. 

>    - [Windows App SDK Runtime 1.0.3](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/downloads). This will install version 1.0.3 if this or newer version is not installed already.

Reviewer: #18603

>    - [Windows App SDK Runtime 1.0.3](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/downloads). This will install version 1.0.3 if this or newer version is not installed already.

Reviewer: ```suggestion
   - [Windows App SDK Runtime 1.1](https://docs.microsoft.com/windows/apps/windows-app-sdk/downloads). This will install version 1.1.0 if this or newer version is not installed already.
```

>    - [Microsoft Visual C++ Redistributable](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022) installer. This will install one of the latest versions available.

Reviewer: ```suggestion
   - [Microsoft Visual C++ Redistributable](https://docs.microsoft.com/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022) installer. This will install one of the latest versions available.
```

### Via GitHub with EXE [Recommended]

>  [Microsoft PowerToys GitHub releases page][github-release-link], click on `Assets` at the bottom to show the files available in the release and then click on `PowerToysSetup-0.59.0-x64.exe` or `PowerToysSetup-0.59.0-arm64.exe` to download the PowerToys installer that matches your machine's architecture. (Note: There's a known issue where PowerToys can't install correctly on Windows 10 on arm64.)

Reviewer: >(Note: There's a known issue where PowerToys can't install correctly on Windows 10 on arm64.)

This looks pretty important. Maybe this doesn't need to be in parentheses and I would prefer a link to issue(s) to see _why_.

Author: There's an entry for the known issue right after. I'll link an issue there.

Author: Removed parenthesis and created an issue.

Our [prioritized roadmap][roadmap] of features and utilities that the core team is focusing on.

### 0.59 - May 2022 Update

In this release, we focused on wrapping up building for native ARM64 and releasing the first experimental build. Below are some of the highlights!

**Highlights**

- The work for running natively on ARM64 has been wrapped up and a build is released. Thanks [@snickler](https://github.com/snickler)!
- Power Rename no longer runs on WinUI 2 and is now running on WinUI 3.
- Keyboard Manager now allows up to 4 modifier keys for shortcuts and has received some quality fixes.

### Known issues

> - There are reports of users who are [unable to open the Settings window](https://github.com/microsoft/PowerToys/issues/18015). This is being caused by incompatibilities with some applications in some cases, or by some UAC settings which cause every application to run elevated. If you're affected by this, please check the  linked issue to verify if any of the presented solutions works for you.

Author: Added the known examples

### General

- Some typos were fixed. Thanks [@eltociear](https://github.com/eltociear), [@rcmaehl](https://github.com/rcmaehl) and [@ShyPixie](https://github.com/ShyPixie)!

### ARM64

- Added ARM64 configurations to the remaining projects.
- Fixed PowerRename to work on ARM64.
- Fixed File Explorer tools to work on ARM64.
- Made changes for the installer projects to build ARM64 installers.
- Configured the CI and Release pipelines to build for ARM64.
- Added ARM64 build status to the README.

### Always on Top

- Fixed an issue where the borders where sticking around when a window was minimized with Win+D.

### FancyZones

- Fixed a bug that was consuming CPU cycles when the default layout was set.
- Fixed a bug where apps were not opened in their last known zones due to Virtual Desktop ID changes.
- Fixed a bug that was snapping popup menus opened by applications.
- Fixed a bug causing windows not to be snapped under some configurations.

### Image Resizer

- No longer tries to change metadata on files that were not actually resized. Thanks [@adamchilders](https://github.com/adamchilders)!

> ### File explorer

Reviewer 2: ```suggestion
### File explorer add-ons
```

> - Fixed a bug where modules depending on WebView2 would be limited to opening files smaller than 2 MB.

Reviewer 3: Not dev file preview.

Author: Dev file preview has a limit of 50kb from the beginning I believe, so it won't hit the 2MB limit, I believe.

> - Fixed a bug where modules depending on WebView2 would be limited to opening files smaller than 2 MB.

Author: Added the known examples

- Add a viewBox attribute to svgs that don't have one so that the preview tries to show the whole image.
- Remove scrollbar that was showing when rendering svg thumbnails.

### Keyboard Manager

- Now up to 4 modifier keys can be used in shortcuts.
- Fixed a bug locking Keyboard Manager when two shortcut mapping were pressed at the same time.
- Removed event spam for certain telemetry events.

### PowerRename

- Ported to use WinUI 3 instead of WinUI 2.

### PowerToys Run

> - The Services plugin is able to search for parts of the name, display name or result type. Thanks [@htcfreek](https://github.com/htcfreek)!

Reviewer 4: ```suggestion
- The Services plugin is able to search for parts of the name, display name or service type. Thanks [@htcfreek](https://github.com/htcfreek)!
- The Services plugin now supports the startup type 'Automatic (Delayed autostart)'. Thanks [@htcfreek](https://github.com/htcfreek)!
```

**Highlights**

- The work for running natively on ARM64 has been wrapped up and a build is released. Thanks [@snickler](https://github.com/snickler)!
- Power Rename no longer runs on WinUI 2 and is now running on WinUI 3.
- Keyboard Manager now allows up to 4 modifier keys for shortcuts and has received some quality fixes.

### Known issues
- There are reports of users who are [unable to open the Settings window](https://github.com/microsoft/PowerToys/issues/18015). This is being caused by incompatibilities with some applications in some cases (RTSS RivaTuner Statistics Server and MSI AfterBurner are known examples of this), or by some UAC settings which cause every application to run elevated. If you're affected by this, please check the  linked issue to verify if any of the presented solutions works for you.
- Some users are [unable to install PowerToys](https://github.com/microsoft/PowerToys/issues/18021) due to the Windows App SDK runtime failing to install. If you're affected by this, please check the issue to verify if any of the presented solutions works for you.

> - PowerToys can't currently install on Windows 10 ARM64, due to the Windows App SDK installer dependency failing to install. We're aware of this and investigating a workaround until the Windows App SDK installer supports Windows 10 ARM64.

Reviewer: So, "The work for running natively on ARM64 has been wrapped up and a build is released", but it can't be installed? That's really confusing...

Author: It can be installed on Windows 11, which seems to be where the majority of ARM64 users of Windows are already. I'll make it clearer.

### ARM64

- Added ARM64 configurations to the remaining projects.
- Fixed PowerRename to work on ARM64.
- Fixed File Explorer tools to work on ARM64.
- Made changes for the installer projects to build ARM64 installers.
- Configured the CI and Release pipelines to build for ARM64.
- Added ARM64 build status to the README.

### Always on Top

- Fixed an issue where the borders where sticking around when a window was minimized with Win+D.

### FancyZones

- Fixed a bug that was consuming CPU cycles when the default layout was set.
- Fixed a bug where apps were not opened in their last known zones due to Virtual Desktop ID changes.
- Fixed a bug that was snapping popup menus opened by applications.
- Fixed a bug causing windows not to be snapped under some configurations.

### Image Resizer

- No longer tries to change metadata on files that were not actually resized. Thanks [@adamchilders](https://github.com/adamchilders)!

### File explorer add-ons

- Fixed a bug where modules depending on WebView2 would be limited to opening files smaller than 2 MB. Now the resulting html is generated into a temporary file before presenting it.

> - Add a viewBox attribute to svgs that don't have one so that the preview tries to show the whole image.

Reviewer: ```suggestion
- Add a viewBox attribute to svg files that don't have one so that the preview tries to show the whole image.
```
(it's a nit, i know)

- Remove scrollbar that was showing when rendering svg thumbnails.

### Keyboard Manager

> - Now up to 4 modifier keys can be used in shortcuts.  This will allow you to use the Office key as an example.

Reviewer: ```suggestion
- Now up to four modifier keys can be used in shortcuts. This will allow you to use the Office key (which sends Win+Ctrl+Shift+Alt), for example.
```

- The Services plugin now supports the startup type 'Automatic (Delayed Autostart)'. Thanks [@htcfreek](https://github.com/htcfreek)!
- The Services plugin now has tooltips for large service names and other UI improvements. Thanks [@htcfreek](https://github.com/htcfreek)!
- The TimeDate plugin gave results for queries containing just numbers on global queries. This has been fixed. Thanks [@htcfreek](https://github.com/htcfreek)!
- We've introduced a throttle before a query is done to ensure typing is done to increase performance. Thanks [@shandsj](https://github.com/shandsj)!
- Fixed a crash in WebSearch when there's an empty pattern setup for the system's default browser.
- Fixed a bug where VSCodeWorkspaces was not finding portable installations of VSCode. Thanks [@harvastum](https://github.com/harvastum)!
- The Calculator plugin reacts better to invalid input and internal errors. Thanks [@htcfreek](https://github.com/htcfreek)!
- The Calculator plugin can now be configured to use the US number format instead of the system one. Thanks [@htcfreek](https://github.com/htcfreek)!

> - The Folder plugin supports paths starting with "/". Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

Reviewer 4: ```suggestion
- The Folder plugin supports paths containing "/". Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
```

- The Services plugin now supports the startup type 'Automatic (Delayed Autostart)'. Thanks [@htcfreek](https://github.com/htcfreek)!
- The Services plugin now has tooltips for large service names and other UI improvements. Thanks [@htcfreek](https://github.com/htcfreek)!
- The TimeDate plugin gave results for queries containing just numbers on global queries. This has been fixed. Thanks [@htcfreek](https://github.com/htcfreek)!
- We've introduced a throttle before a query is done to ensure typing is done to increase performance. Thanks [@shandsj](https://github.com/shandsj)!
- Fixed a crash in WebSearch when there's an empty pattern setup for the system's default browser.
- Fixed a bug where VSCodeWorkspaces was not finding portable installations of VSCode. Thanks [@harvastum](https://github.com/harvastum)!
- The Calculator plugin reacts better to invalid input and internal errors. Thanks [@htcfreek](https://github.com/htcfreek)!
- The Calculator plugin can now be configured to use the US number format instead of the system one. Thanks [@htcfreek](https://github.com/htcfreek)!

> - The Folder plugin supports paths containing "/". Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

Reviewer 3: ```suggestion
- The Folder plugin now supports paths starting with "/". Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
```

Reviewer 4: No. Containing with. @davidegoacometti fixed two issues.
